ddtrace/tracer/stats.go

```diff
@@ -178,15 +176,24 @@ func (c *concentrator) flush(timenow time.Time) statsPayload {
 		Stats:    make([]statsBucket, 0, len(c.buckets)),
 	}
 	for ts, srb := range c.buckets {
-		if ts > now-c.bucketSize {
+		if !includeCurrent && ts > now-c.bucketSize {
 			// do not flush the current bucket
 			continue
 		}
 		log.Debug("Flushing bucket %d", ts)
 		sp.Stats = append(sp.Stats, srb.Export())
 		delete(c.buckets, ts)
 	}
-	return sp
+	if len(sp.Stats) == 0 {
+		// nothing to flush
+		return
+	}
+	c.statsd().Incr("datadog.tracer.stats.flush_payloads", nil, 1)
+	c.statsd().Incr("datadog.tracer.stats.flush_buckets", nil, float64(len(sp.Stats)))
+	if err := c.cfg.transport.sendStats(&sp); err != nil {
```

Comment:
Moving the sendStats here guards it with `c.mu` which wasn't the case before, which could have performance implications.

Instead, I'm going to move this to a helper function